### PR TITLE
app preview improvements

### PR DESCRIPTION
**Pull request by mjashanks**
## Description
1. App preview does not do a full reload on property change
2. When a screen is selected, it is displayed within the master page
3. Disable POST, PUT, DELETE api calls when in builder preview

#343

### CODE REVIEW

File: packages/builder/src/components/userInterface/AppPreview/iframeTemplate.js [hunk 1]
```diff
@@ -35,12 +21,45 @@ export default ({
         }
     </style>
     <script>
-        window["##BUDIBASE_FRONTEND_DEFINITION##"] = ${frontendDefinition};
+      function receiveMessage(event) {
```
Comment (shogunpurple) [inline packages/builder/src/components/userInterface/AppPreview/iframeTemplate.js:24]: NAB: You can avoid adding this function as a string by doing something like this in the parent svelte component:

```js
iframe.contentWindow.receiveMessage = function() {
....
}
```

**Review by shogunpurple — APPROVED** [review]
Just a few optional things. Nice work 🚀

File: packages/builder/src/components/userInterface/AppPreview/CurrentItemPreview.svelte [hunk 2]
```diff
@@ -103,6 +100,25 @@
...
+      styles,
+      stylesheetLinks,
+      selectedComponentType,
+      selectedComponentId,
+      frontendDefinition,
+    }))
+  }
+  let iframeLoaded = false
+  $: if(iframe && !iframeLoaded) {
+    iframe.contentWindow.addEventListener("bb-ready", refreshContent)
```
[the remark this answers is not in the document]
Comment (mjashanks) [reply packages/builder/src/components/userInterface/AppPreview/CurrentItemPreview.svelte:115]: nice! did not know that

**Comment by mjashanks** [comment]
@shogunpurple  i added a new commit to this.... for dsiabling POST, PUT, DELETE in preview

File: packages/client/src/api/index.js [hunk 3]
```diff
@@ -30,10 +35,12 @@ export const createApi = ({ rootPath = "", setState, getState }) => {
...
   }
 
-  const post = apiCall("POST")
+  const dummyCall = () => ({})
+
+  const post = isBuilderPreview ? dummyCall : apiCall("POST")
   const get = apiCall("GET")
-  const patch = apiCall("PATCH")
-  const del = apiCall("DELETE")
+  const patch = isBuilderPreview ? dummyCall : apiCall("PATCH")
```
Comment (shogunpurple) [inline packages/client/src/api/index.js:42]: What we are intending to do in the builder preview is to disable click events. We can remove all the `isBuilderPreview` stuff here, and simply disable click events inside the iframe, which should give us the same result:

```js
iframe.onclick = function(e) {
                e.preventDefault();
                return false;
}
```

should be all the code you need.

**Pull request merged**